### PR TITLE
feat: gitea app operator

### PR DESCRIPTION
**Pull request by ferruhcihan**
### Implements:
https://jira.linode.com/browse/APL-23
https://jira.linode.com/browse/APL-13

### Description:
This PR updates the `gitea-operator` files and eliminates the `gitea-prepare` job, integrating its tasks into the `gitea-operator` for improved task management.
Is paired with: https://github.com/redkubes/otomi-tasks/pull/103

## Checklist

- [ ] Architecture Design Records have been added as `adr/*.md` and appended to list in `adr/_index.md`, if applicable.
- [ ] The `values-schema.yaml` file and `test/**` fixtures have been updated to reflect code changes, if applicable.
- [ ] The OpenApi Schema from redkubes/otomi-api project is compatible with definitions from `values-schema.yaml` file, if applicable.
- [ ] Helm releases are meeting otomi's baseline security policies, if applicable.
- [ ] Helm chart and helmfile changes are tested against upgrade scenario, if applicable.

### CODE REVIEW

**Review by CasLubbers — APPROVED** [review]

File: charts/gitea-app-operator/templates/tests/test-connection.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
```
Comment (j-zimnowoda) [inline charts/gitea-app-operator/templates/tests/test-connection.yaml:1]: operator is not a server

File: charts/gitea-app-operator/templates/rbac.yaml [hunk 2]
```diff
@@ -2,35 +2,41 @@
...
+  name: {{ include "gitea-app-operator.serviceAccountName" . }}
   namespace: {{ .Release.Namespace | quote }}
 automountServiceAccountToken: true
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
-  name: {{ include "gitea-operator.fullname" . }}-edit-oauth
+  name: {{ include "gitea-app-operator.fullname" . }}-edit-oauth
 rules:
```
Comment (j-zimnowoda) [inline charts/gitea-app-operator/templates/rbac.yaml:13]: Could you justify (as a comment in this file) each RBAC rule?

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
Looks good, just few minor remarks

File: core.yaml [hunk 3]
```diff
@@ -24,7 +24,7 @@ k8s:
     - name: harbor
       app: harbor
     - name: gitea
-    - name: gitea-operator
+    - name: gitea-app
```
[the remark this answers is not in the document]
Comment (ferruhcihan) [reply core.yaml]: Operator name and namespace are updated!

File: charts/gitea-app-operator/templates/rbac.yaml (hunk 2 above)
Comment (ferruhcihan) [reply charts/gitea-app-operator/templates/rbac.yaml]: Comment lines are added!

**Review by j-zimnowoda — APPROVED** [review]
Well done. thanks for adding the comments!

**Pull request merged**